proxy/codecs.go

```diff
@@ -19,13 +19,14 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"io/ioutil"
 
 	"github.com/datastax/go-cassandra-native-protocol/frame"
 	"github.com/datastax/go-cassandra-native-protocol/message"
 	"github.com/datastax/go-cassandra-native-protocol/primitive"
 )
 
-var codec = frame.NewRawCodec(&partialQueryCodec{}, &partialExecuteCodec{})
+var codec = frame.NewRawCodec(&partialQueryCodec{}, &partialExecuteCodec{}, &partialBatchCodec{})
```

Comment:
What is the reason for these partial codecs? I assume that it has to do with avoiding decoding/encoding parts of the message that the proxy doesn't care about? If this is the case, then the proxy can only use these codecs to inspect the frames but not to modify them, right?

Reply:
>  I assume that it has to do with avoiding decoding/encoding parts of the message that the proxy doesn't care about?

Exactly this. 

> If this is the case, then the proxy can only use these codecs to inspect the frames but not to modify them, right?

Correct, cql-proxy on inspects and doesn't currently modify frames. There hasn't been a need yet.